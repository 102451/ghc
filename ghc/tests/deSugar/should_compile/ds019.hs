--!!! ds019 -- mixed var and uni-constructor pats

module Test where

f (a,b,c) i     o = []
f d       (j,k) p = []
f (e,f,g) l     q = []
f h       (m,n) r = []
