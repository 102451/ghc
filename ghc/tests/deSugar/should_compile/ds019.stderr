ds019.hs:7: 
    Warning: Pattern match(es) completely overlapped
	in the definition of function `f'
ds019.hs:8: 
    Warning: Pattern match(es) completely overlapped
	in the definition of function `f'


================================================================================
Desugared:
Rec {
f{-r3y,x-}  ::  _forall_
		[tazA taz7 taza tazd tazq tazt tazy]
		=>
		(taz7, taza, tazd) -> (tazq, tazt) -> tazA -> [tazy]
{-# L #-}
f{-r3y,x-} =
    _/\_ tazA taz7 taza tazd tazq tazt tazy ->
	_letrec_ {
	  f_ayl  ::  (taz7, taza, tazd) -> (tazq, tazt) -> tazA -> [tazy]
	  {-# L #-}
	  f_ayl =
	      \ ds_dEd  ::  (taz7, taza, tazd)
		{-# L #-}
		ds_dEd i_r3g  ::  (tazq, tazt)
		       {-# L #-}
		       i_r3g o_r3h  ::  tazA
			     {-# L #-}
			     o_r3h ->
		  case ds_dEd of { PrelTup.(,,){-63,w-}{i} a_r3d b_r3e c_r3f  ->
		  PrelBase.[]{-5i,w-}{i} _@_ tazy;};
	} in  f_ayl
end Rec }
